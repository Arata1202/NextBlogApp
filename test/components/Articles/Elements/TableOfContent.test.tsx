import { render } from '@testing-library/react';
import TableOfContents from '@/components/Articles/Elements/TableOfContent';

const mockHeadings = [
  { id: '1', title: 'Heading 1', level: 1 },
  { id: '2', title: 'Heading 2', level: 2 },
  { id: '3', title: 'Heading 3', level: 3 },
  { id: '4', title: 'Heading 2', level: 2 },
  { id: '5', title: 'Heading 1', level: 1 },
];

describe('TableOfContent', () => {
  test('スナップショット（TableOfContent）', () => {
    const { asFragment } = render(<TableOfContents headings={mockHeadings} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
