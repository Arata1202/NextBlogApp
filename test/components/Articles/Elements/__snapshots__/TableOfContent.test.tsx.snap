// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableOfContent スナップショット（TableOfContent） 1`] = `
<DocumentFragment>
  <div>
    <div
      class="flex justify-center"
    >
      <nav
        aria-label="Table of contents"
        class="tableOfContent w-1/2 border p-4 LightTheme"
      >
        <h1
          class="text-center font-bold text-lg"
        >
          目次
        </h1>
        <ol
          class="mt-4 list-none pl-0"
        >
          <li
            style="margin-left: 0px; background-color: transparent; transition: background-color 0.3s ease;"
          >
            <a
              class="hover:text-blue-500"
              href="#1"
            >
              1 Heading 1
            </a>
          </li>
          <li
            style="margin-left: 20px; background-color: transparent; transition: background-color 0.3s ease;"
          >
            <a
              class="hover:text-blue-500"
              href="#2"
            >
              1.1 Heading 2
            </a>
          </li>
          <li
            style="margin-left: 40px; background-color: transparent; transition: background-color 0.3s ease;"
          >
            <a
              class="hover:text-blue-500"
              href="#3"
            >
              1.1.1 Heading 3
            </a>
          </li>
          <li
            style="margin-left: 20px; background-color: transparent; transition: background-color 0.3s ease;"
          >
            <a
              class="hover:text-blue-500"
              href="#4"
            >
              1.2 Heading 2
            </a>
          </li>
          <li
            style="margin-left: 0px; background-color: transparent; transition: background-color 0.3s ease;"
          >
            <a
              class="hover:text-blue-500"
              href="#5"
            >
              2 Heading 1
            </a>
          </li>
        </ol>
      </nav>
    </div>
  </div>
</DocumentFragment>
`;
